code/modules/clothing/shoes/miscellaneous.dm

```diff
@@ -18,6 +18,24 @@
 	permeability_coefficient = 0.05 //Thick soles, and covers the ankle
 	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes
 
+/obj/item/clothing/shoes/combat/sneakboots
+	name = "sneakboots"
+	desc = "These boots have special noise cancelling soles. Perfect for stealth, if it wasn't for the color scheme."
+	icon_state = "sneakboots"
+	item_state = "sneakboots"
+	resistance_flags = FIRE_PROOF |  ACID_PROOF
+
+/obj/item/clothing/shoes/combat/sneakboots/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(slot != ITEM_SLOT_FEET)
+		return
+	if(slot == ITEM_SLOT_FEET)
+		ADD_TRAIT(user, TRAIT_SILENT_FOOTSTEPS, SHOES_TRAIT)
+
+/obj/item/clothing/shoes/combat/sneakboots/dropped(mob/living/carbon/human/user)
+	. = ..()
+	REMOVE_TRAIT(user, TRAIT_SILENT_FOOTSTEPS, SHOES_TRAIT)
```

Comment:
You might want to perform the action first and later call parent with `return ..()` on removal/deleting procs in general, as parent calls may do some cleanup actions changing the state of the item.
As in, replace `. = ..()` in the beginning with `return ..()`